done/done.py

```diff
@@ -1,77 +1,110 @@
 from xblock.fragment import Fragment
 
+try:
+    from eventtracking import tracker
+except ImportError:
+    class tracker(object):  # pylint: disable=invalid-name
+        """
+        Define tracker if eventtracking cannot be imported. This is a
+        workaround so that the code works in both edx-platform and
+        XBlock workbench (the latter of which does not support event
+        emission). This should be replaced with XBlock's emit(), but
+        at present, emit() is broken.
+        """
+        def __init__(self):
+            """ Do nothing """
+            pass
+
+        @staticmethod
+        def emit(param1, param2):
+            """ In workbench, do nothing for event emission """
+            pass
+
+
+def resource_string(path):
+    """Handy helper for getting resources from our kit."""
+    data = pkg_resources.resource_string(__name__, path)
+    return data.decode("utf8")
+
 
 class DoneXBlock(XBlock):
     """
     Show a toggle which lets students mark things as done.
     """
 
-    # Fields are defined on the class.  You can access them in your code as
-    # self.<fieldname>.
     done = Boolean(
-           scope = Scope.user_state, 
-           help = "Is the student done?",
-           default = False
-        )
+        scope=Scope.user_state,
+        help="Is the student done?",
+        default=False
+    )
 
     align = String(
-           scope = Scope.settings, 
-           help = "Align left/right/center",
-           default = "left"
-        )
+        scope=Scope.settings,
+        help="Align left/right/center",
+        default="left"
+    )
 
     has_score = True
 
-    def resource_string(self, path):
-        """Handy helper for getting resources from our kit."""
-        data = pkg_resources.resource_string(__name__, path)
-        return data.decode("utf8")
-
+    # pylint: disable=unused-argument
     @XBlock.json_handler
     def toggle_button(self, data, suffix=''):
-        self.done = data['done']
-        if data['done']:
-            grade = 1
-        else:
-            grade = 0
-
-        self.runtime.publish(self, 'grade', {'value':grade, 'max_value': 1})
-        return {}
-
-
-    def student_view(self, context=None):
+        """
+        Ajax call when the button is clicked. Input is a JSON dictionary
+        with one boolean field: `done`. This will save this in the
+        XBlock field, and then issue an appropriate grade.
+        """
+        if 'done' in data:
+            self.done = data['done']
+            if data['done']:
+                grade = 1
+            else:
+                grade = 0
+            grade_event = {'value': grade, 'max_value': 1}
+            self.runtime.publish(self, 'grade', grade_event)
+            # This should move to self.runtime.publish, once that pipeline
+            # is finished for XBlocks.
+            tracker.emit("edx.done.toggle", {'done': self.done})        
```

Comment:
@stroilova The events are identical and happen within a few milliseconds of each other. The only differences are that the failure modes are different. Client-side events cannot be trusted (e.g. they can be spoofed). The client-side events can be subscribed to by other XBlocks, and aren't effected by e.g. some types of server errors and similar. They also have (or ought to have) a client-side time stamp. They're also a nice backup on unreliable Internet connections. 

My personal preference is to use identical names in a situation like this -- or even when not semantically different, but it's the same event, and the client and server just have different bits of data. In some cases, only the server knows the final grade, but only the client knows e.g. exact pixels where a user clicked. In that case, it's sensible to merge such events in post-processing, and identical names are nice. 

However, it's unimportant, so I am happy to move on with whatever event names you would like. Please let me know how to proceed. 

I created a wiki pages, except for actual event. I am waiting for devstack to do that (https://openedx.atlassian.net/wiki/display/AN/DoneXBlock). 

One more minor point: This has been out as a prototype for a while now (in very limited roll-outs). If we do change event names, it will add a small bit of inconsistency. This isn't a big deal (very small number of courses), but just something to be aware of.